File: service/app/src/main/java/io/miragon/blueprint/adapter/inbound/operaton/SendRejectionWorker.java
package io.miragon.blueprint.adapter.inbound.operaton;

import dev.bpmcrafters.processengine.worker.ProcessEngineWorker;
import dev.bpmcrafters.processengine.worker.Variable;
import io.miragon.blueprint.adapter.process.BikeLeasingProcessProcessApi.ServiceTasks;
import io.miragon.blueprint.application.port.inbound.RejectApplicationUseCase;
import io.miragon.blueprint.domain.leasing.ApplicationId;
import org.springframework.stereotype.Component;

@Component
public class SendRejectionWorker {

    private final RejectApplicationUseCase useCase;

    public SendRejectionWorker(RejectApplicationUseCase useCase) {
        this.useCase = useCase;
    }

    @ProcessEngineWorker(topic = ServiceTasks.SEND_REJECTION)
    public void sendRejection(@Variable String applicationId) {
        useCase.reject(ApplicationId.of(applicationId));
    }
}

File: service/app/src/main/kotlin/io/miragon/blueprint/adapter/inbound/operaton/SendRejectionWorker.kt
package io.miragon.blueprint.adapter.inbound.operaton

import dev.bpmcrafters.processengine.worker.ProcessEngineWorker
import dev.bpmcrafters.processengine.worker.Variable
import io.miragon.blueprint.adapter.process.BikeLeasingProcessProcessApi.ServiceTasks
import io.miragon.blueprint.application.port.inbound.RejectApplicationUseCase
import io.miragon.blueprint.domain.leasing.ApplicationId
import org.springframework.stereotype.Component

@Component
class SendRejectionWorker(
    private val useCase: RejectApplicationUseCase,
) {

    @ProcessEngineWorker(topic = ServiceTasks.SEND_REJECTION)
    fun sendRejection(@Variable applicationId: String) {
        useCase.reject(ApplicationId.of(applicationId))
    }
}
